addons/base_geolocalize/models/res_partner.py

```diff
@@ -5,6 +5,17 @@ class ResPartner(models.Model):
     _inherit = "res.partner"
 
     date_localization = fields.Date(string='Geolocation Date')
+    partner_latitude = fields.Float(compute="_compute_geo_coordinates", readonly=False, store=True)
+    partner_longitude = fields.Float(compute="_compute_geo_coordinates", readonly=False, store=True)
+
+    @api.depends('street', 'zip', 'city', 'state_id', 'country_id')
+    def _compute_geo_coordinates(self):
+        for partner in self:
+            if not (partner.city and partner.country_id):
+                partner.partner_latitude = 0.0
+                partner.partner_longitude = 0.0
+            else:
+                partner.geo_localize()
```

Comment:
@tivisse @d-fence  This `@depends` is very dangerous because it will trigger a lot of calls to the GeoIP API. Could we please make it lazy instead?
I mean just reset to 0, and the user can manually trigger the geolocation again when needed. If FSM needs the geolocation, it can trigger it as needed when it's not yet done.

This will also make #84822 useless.